pkg/util/parameters.go

```diff
@@ -0,0 +1,180 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+)
+
+var (
+	errUnknownParameter      = fmt.Errorf("unknown parameter")
+	errUnmodifiableParameter = fmt.Errorf("parameter is not modifiable")
+)
+
+func NewErrUnknownParameter(name string) error {
+	// This is a terminal error because unless the user removes this parameter
+	// from their list of parameter overrides, we will not be able to get the
+	// resource into a synced state.
+	return ackerr.NewTerminalError(
+		fmt.Errorf("%w: %s", errUnknownParameter, name),
+	)
+}
+
+func NewErrUnmodifiableParameter(name string) error {
+	// This is a terminal error because unless the user removes this parameter
+	// from their list of parameter overrides, we will not be able to get the
+	// resource into a synced state.
+	return ackerr.NewTerminalError(
+		fmt.Errorf("%w: %s", errUnmodifiableParameter, name),
+	)
+}
+
+// ComputeParametersDelta compares two Parameter arrays and returns the new
+// parameters to add, to update and the parameter identifiers to delete
+func ComputeParametersDelta(
+	desired map[string]*string,
+	latest map[string]*string,
+) (map[string]*string, []string) {
+	toReset := []string{}
+	toModify := map[string]*string{}
+
+	for k, v := range desired {
+		if lv, found := latest[k]; !found {
+			toModify[k] = v
+		} else if !equalStrings(v, lv) {
+			toModify[k] = v
+		}
+	}
+	for k := range latest {
+		if _, found := desired[k]; !found {
+			toReset = append(toReset, k)
+		}
+	}
+	return toModify, toReset
+}
+
+// sliceStringChunks splits a supplied slice of string pointers into multiple
+// slices of string pointers of a given size.
+func SliceStringChunks(
```

Comment:
`samber/lo` supports this out of the box - https://github.com/samber/lo#chunk

Reply:
Done